test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -260,6 +260,46 @@ TEST_F(ClusterManagerImplTest, UnknownHcType) {
   EXPECT_THROW(create(*loader), EnvoyException);
 }
 
+TEST_F(ClusterManagerImplTest, MaxClusterName) {
+  std::string json = R"EOF(
+  {
+    "clusters": [
+    {
+      "name": "clusterwithareallyreallylongnamemorethanmaxcharsallowedbyschema"
+    }]
+  }
+  )EOF";
+
+  Json::ObjectPtr loader = Json::Factory::LoadFromString(json);
+  try {
+    create(*loader);
+  } catch (Json::Exception& e) {
+    EXPECT_EQ("JSON object doesn't conform to schema.\n Invalid schema: #/properties/name.\n "
```

Comment:
I disagree, this is a better test. Otherwise a JSON exception could be thrown for any number of reasons. If you don't want to do an exact string match, you could do a find(), or even change the Exception type to be a JsonSchemaException which derives from JsonException.

Also, as an aside, you need to fail in the case where you don't throw an exception. Optimally you would be doing something like EXPECT_THROW(). It would be nice to actually make a variant of EXPECT_THROW() that can check against a message also.